tools/lib/tests/test_logreader.py

```diff
@@ -118,6 +118,12 @@ def test_multiple_iterations(self, init_mock):
 
     self.assertEqual(qlog_len1, qlog_len2)
 
+  @pytest.mark.slow
```

Comment:
test to make sure it remains an iterable? .first('carParams') should never download/parse past the first qlog imo

Reply:
> remains an iterable
as in you can still iterate after .first?

> .first('carParams') should never download/parse past the first qlog imo
hmm, even if it fails to find it within the first log?

Reply:
That's fine, I meant it doesn't return a list